### PR TITLE
Added StringName as a variant type.

### PR DESCRIPTION
**Pull request by reduz**
Also changed all relevant properties defined manually to StringName.

### CODE REVIEW

File: core/method_ptrcall.h [hunk 1]
```diff
@@ -418,7 +419,7 @@ MAKE_VECARR(Plane);
 		}                                                                     \
 	}
 
-MAKE_STRINGCONV(StringName);
+//MAKE_STRINGCONV(StringName);
```
Comment (akien-mga) [inline core/method_ptrcall.h]: Better remove than comment out.

File: core/global_constants.cpp [hunk 2]
```diff
@@ -606,6 +606,7 @@ void register_global_constants() {
 	BIND_GLOBAL_ENUM_CONSTANT_CUSTOM("TYPE_BASIS", Variant::BASIS);
 	BIND_GLOBAL_ENUM_CONSTANT_CUSTOM("TYPE_TRANSFORM", Variant::TRANSFORM);
 	BIND_GLOBAL_ENUM_CONSTANT_CUSTOM("TYPE_COLOR", Variant::COLOR);
+	BIND_GLOBAL_ENUM_CONSTANT_CUSTOM("TYPE_STRING_NAME", Variant::STRING_NAME); // 15
 	BIND_GLOBAL_ENUM_CONSTANT_CUSTOM("TYPE_NODE_PATH", Variant::NODE_PATH); // 15
```
Comment (volzhs) [inline core/global_constants.cpp]: should `// 15` be removed then? :smile:

**Comment by aaronfranke** [comment]
Why? Not trying to question this change, I'm just curious.

File: core/global_constants.cpp (hunk 2 above)
Comment (aaronfranke) [reply core/global_constants.cpp]: Is there any reason to keep these comments at all anyway? Quite a lot of them aren't even in the right places anymore.

**Comment by reduz** [comment]
@aaronfranke A lot of functinos in the Godot API take StringNames because they are much faster to compare and process than regular strings (yet they can be identified as such). Examples are signal names, method names, property names, class names, animation names, shader parameter names, skeleton bone names,  translations keys, etc (the list is very  long actually). They are exposed as StringName in the C++ API but the script API only sees them as simple strings.

By not having this datatype, the script API is forced to convert to StringName every time a function is called, which involves hashing and entire string comparisons every time, significantly reducing performance.

StringNames are effectively a different data type with a different use case, because they are meant to have a textual meaning, but  be mostly compared, hashed, etc. quickly. They also can't be used for string operations, because that is not their purpose (you need to convert them to regular strings first).

File: core/global_constants.cpp (hunk 2 above)
Comment (akien-mga) [reply core/global_constants.cpp]: I guess it might be best to drop them yeah if we can't keep them in sync.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**